### PR TITLE
Cu 8685uwxba council add while runner

### PR DESCRIPTION
**Pull request by Naccomy**

### CODE REVIEW

**Comment by aflament** [comment]
Task linked: [CU-8685uwxba [Council] add While runner](https://app.clickup.com/t/8685uwxba)

File: council/contexts/_budget.py [hunk 1]
```diff
@@ -125,7 +125,7 @@ def is_expired(self) -> bool:
         if self._deadline < time.monotonic():
             return True
 
-        return any(limit.value < 0.0 for limit in self._remaining)
+        return any(limit.value <= 0.0 for limit in self._remaining)
```
Comment (Naccomy) [inline council/contexts/_budget.py:128]: @gkoch78 / @aflament I updated this line to ensure the budget is expired as soon as one of its consumption limits reach 0. Does that make sense to you?

**Review by ToJen — APPROVED** [review]

**Review by gkoch78 — CHANGES_REQUESTED** [review]
we may want to add it in the class diagram : https://council--95.org.readthedocs.build/en/95/reference/runners.html

File: council/runners/while_runner.py [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+class While(RunnerBase):
+    """
+    Runner that executes while the given predicates return `True`
+    """
+
+    def __init__(self, predicate: RunnerPredicate, runner: RunnerBase):
+        """
+        Args:
+            predicate: a predicate function
```
Comment (aflament) [inline council/runners/while_runner.py:13]:
```suggestion
            predicate: a predicate function evaluated in the while loop
```

**Pull request merged**